test/unit/alphabet/semi_alphabet_constexpr_test_template.hpp

```diff
@@ -0,0 +1,159 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alphabet/concept.hpp>
+
+using namespace seqan3;
+
+template <typename T>
+using semi_alphabet_constexpr = ::testing::Test;
+
+TYPED_TEST_CASE_P(semi_alphabet_constexpr);
+
+TYPED_TEST_P(semi_alphabet_constexpr, concept_check)
+{
+    EXPECT_TRUE(detail::constexpr_semialphabet<TypeParam>);
+    EXPECT_TRUE(detail::constexpr_semialphabet<TypeParam &>);
+
+    EXPECT_TRUE(detail::constexpr_semialphabet<TypeParam const>);
+    EXPECT_TRUE(detail::constexpr_semialphabet<TypeParam const &>);
+
+    EXPECT_TRUE(detail::writable_constexpr_semialphabet<TypeParam>);
+    EXPECT_TRUE(detail::writable_constexpr_semialphabet<TypeParam &>);
+
+    EXPECT_FALSE(detail::writable_constexpr_semialphabet<TypeParam const>);
+    EXPECT_FALSE(detail::writable_constexpr_semialphabet<TypeParam const &>);
+}
+
+TYPED_TEST_P(semi_alphabet_constexpr, default_value_constructor)
+{
+    [[maybe_unused]] constexpr TypeParam t0{};
+}
+
+TYPED_TEST_P(semi_alphabet_constexpr, global_assign_rank)
+{
+    constexpr size_t rank = 1 % alphabet_size<TypeParam>;
+    [[maybe_unused]] constexpr TypeParam t0{seqan3::assign_rank_to(rank, TypeParam{})};
```

Comment:
```suggestion
    constexpr TypeParam t0{seqan3::assign_rank_to(rank, TypeParam{})};
    EXPECT_EQ(t0, TypeParam{rank});
```

Reply:
You cannot directly construct an alphabet instance from its rank. There is no way we can test the value except with `to_rank`